src/main/java/com/zufar/onlinestore/user/exception/handler/UserExceptionHandler.java

```diff
@@ -0,0 +1,26 @@
+package com.zufar.onlinestore.user.exception.handler;
+
+import com.zufar.onlinestore.common.exception.handler.GlobalExceptionHandler;
+import com.zufar.onlinestore.common.response.ApiResponse;
+import com.zufar.onlinestore.payment.exception.PaymentNotFoundException;
+import com.zufar.onlinestore.user.exception.UserAlreadyRegisteredException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@Slf4j
+@RestControllerAdvice
+public class UserExceptionHandler extends GlobalExceptionHandler {
```

Comment:
I think this exception is related to security model more than to user model. That is why I think you should rename this class like `SecurityExceptionHandler` and move this class to `com.zufar.onlinestore.security.exception.handler` package